doc/tutorials/dnn/dnn_text_spotting/dnn_text_spotting.markdown

```diff
@@ -0,0 +1,366 @@
+sha: 3940942b85761c7f240494cf662dcbf05dc00d14
+alphabet_3944.txt: https://drive.google.com/uc?export=dowload&id=18IZUUdNzJ44heWTndDO6NNfIpJMmN-ul
+parameter setting: -rgb=1;
+description: The classification number of this model is 3944 (0~9 + a~z + A~Z + Chinese characters + special characters).
+             The training dataet is ReCTS (https://rrc.cvc.uab.es/?ch=12).
+```
+
+#### TextDetectionModel:
+```
+- DB_IC15_resnet50.onnx:
+url: https://drive.google.com/uc?export=dowload&id=1cmAs91yFRyC2LC3nTTBNNYooyEmb0i5t
+sha: 9a562c2a8fa3d95fbecd2d45690fb4f787fd496a
+parameter setting: -inputHeight=736, -inputWidth=1280;
+description: This model is trained on ICDAR2015, so it is only able to detect English text instances.
+
+- DB_IC15_resnet18.onnx:
+url: https://drive.google.com/uc?export=dowload&id=1JQ_-R7_feD25qE2uibpKM1S-R3keTD1U
+sha: b215b5fb85d1c606d1f56f03e8c841025e72733d
+parameter setting: -inputHeight=736, -inputWidth=1280;
+description: This model is trained on ICDAR2015, so it is only able to detect English text instances.
+
+- DB_TD500_resnet50.onnx:
+url: https://drive.google.com/uc?export=dowload&id=1_9KB6QkIghp2VNZti_dJ3Araz-nDZBsL
+sha: 6d3b7b84fc23906be7d583a8cedef727cc351fad
+parameter setting: -inputHeight=736, -inputWidth=736;
+description: This model is trained on MSRA-TD500, so it is able to detect English and Chinese text instances.
+
+- DB_TD500_resnet18.onnx:
+url: https://drive.google.com/uc?export=dowload&id=1f0dfqG_-Fiq4ywdrRhq8JH9PXCcJDO01
+sha: dd19abe7e2326ac660ce2733be1cb7032f76d458
+parameter setting: -inputHeight=736, -inputWidth=736;
+description: This model is trained on MSRA-TD500, so it is able to detect English and Chinese text instances.
+
+```
+
+The models with Res50 have a better accuracy, while those with Res18 have a faster inference speed.
+
+## Images for Testing
+```
+Text Recognition:
+url: https://drive.google.com/uc?export=dowload&id=1nMcEy68zDNpIlqAn6xCk_kYcUTIeSOtN
+sha: 89205612ce8dd2251effa16609342b69bff67ca3
+
+Text Detection:
+url: https://drive.google.com/uc?export=dowload&id=149tAhIcvfCYeyufRoZ9tmc2mZDKE_XrF
+sha: ced3c03fb7f8d9608169a913acf7e7b93e07109b
+```
+
+## Example for Text Recognition
+
+```cpp
+#include <iostream>
+#include <fstream>
+
+#include <opencv2/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/dnn.hpp>
+
+using namespace cv;
+using namespace dnn;
+
+int main()
+{
+    // Load model weights
+    TextRecognitionModel model("path/to/crnn_cs.onnx");
+
+    // Load vocabulary
+    // vocabulary should be changed according to the text recognition model
+    std::ifstream vocFile;
+    vocFile.open("path/to/alphabet_94.txt");
+    CV_Assert(vocFile.is_open());
+    String vocLine;
+    std::vector<String> vocabulary;
+    while (std::getline(vocFile, vocLine)) {
+        vocabulary.push_back(vocLine);
+    }
+
+    // Normalization parameters
+    double scale = 1.0 / 127.5;
+    Scalar mean = Scalar(127.5);
```

Comment:
`127.5, 0, 0, 0`

or

`127.5, 127.5, 127.5, 127.5`

?

Reply:
I am not sure about your confusion here.
When I specify the `mean` parameter to `Scalar(127.5)`, all the values of the input image will subtract the mean (127.5) in the normalization step.

Reply:
`Scalar(127.5)` only initializes the 1st element, so that would be (127.5, 0, 0, 0)

Reply:
Thank you.
I have tested it and changed it to `Scalar(127.5, 127.5, 127.5)`.